test/cpp/tensorexpr/test_expr.cpp

```diff
@@ -254,6 +254,46 @@ void testExprCompareSelectEQ() {
   assertAllEqual(c_buffer, 1);
 }
 
+void testExprCompareSelectDtypes() {
```

Comment:
Can we have a comment about this test, as you found out offline I thought you'd implemented something different. It's not obvious that the important part is the difference between the type of the condition and value.